frigate/config.py

```diff
@@ -816,12 +817,16 @@ def runtime_config(self) -> FrigateConfig:
             exclude_unset=True,
         )
 
-        for name, camera in config.cameras.items():
+        for name, camera in config.cameras.copy().items():
             merged_config = deep_merge(camera.dict(exclude_unset=True), global_config)
             camera_config: CameraConfig = CameraConfig.parse_obj(
                 {"name": name, **merged_config}
             )
 
+            if not camera_config.enabled:
+                config.cameras.pop(name)
```

Comment:
I haven't tested this, but won't this essentially just prevent the camera's config from being loaded at all? I thought the goal was to make it possible to have retain settings for a disabled camera that are different than the default values.

Reply:
Yeah you're right, I wasn't sure if that was supposed to be in this initial pass. 

In that case we'd want to leave the config itself alone and not run the ffmpeg commands if the camera is disabled.

Reply:
I think we talk about 3 different requirements:
1. static: Option in config, only used one time when starting the container (the original requirement for this PR). Camera settings aren't loaded in config.py.
2. dynamic: Camera can be disabled or enabled while running frigate, so the config is changed in the UI. State of the camera will be saved "somewhere else" but not in the config. Settings are always available.
3. "A bit of both": Option in config, when starting frigate the settings are loaded but the views (recording/events/live/rtmp etc) do not show a disabled camera. But additionaly features in the UI to enable/disable every camera. Setting are always available.

Is it enough to "not run the ffmpeg commands" or is more logic necessary in the different views when a camera isn't shown (example: Events View, dropdown Filter)? I will start digging and trying here today. 

Close this pull request and make a new one based on requirement 2 or 3?
Or update this one and switch to the dynamic solution?

If starting to change parameters dynamically (enabled/disabled is only a first possibility) shouldn't there be a view for configurations showing each camera with a list of parameters that might be changed in the UI (resolution, framerate, etc.)?

Reply:
The dynamic solution should come later, that's not what's being suggested. 

This solution should still be static. The thing that's being discussed is we don't want the cameras config completely removed as we still want the retention settings to be applied. If a camera is disabled it just shouldn't be running, it should still show in the dropdowns and other places because you may still want to view those events

Reply:
Ok. But this is a little bit difficult to understand:
If I disable a camera in the top of the camera config I wouldn't expect parts of the camera still "working" in the UI.

Does "not running" mean the following? 
- no motion, detect, recording, snapshot
- no live view and no debug view
- everything else would still be visible and usable: viewing events, snapshots and records.

I'll admit I have no idea how to solve this right now.

Reply:
> Does "not running" mean the following?
> 
> * no motion, detect, recording, snapshot
> * no live view and no debug view
> * everything else would still be visible and usable: viewing events, snapshots and records.

Yes preciesely, disabled will mean that the ffmpeg / video processes won't run but the camera will otherwise still be in the UI. It should just consist of not starting a process around https://github.com/blakeblackshear/frigate/blob/dev/frigate/app.py#L250 and probably some other places as well

Reply:
Thanks! I started looking already in app.py. And "some other places". I will try it and I hope it won't get too annoying if my commits aren't perfect...

Reply:
Not starting the process when the camera is disabled is working already.
Birdseye, Events, Debug all ok.
In Cameras: I hate the green image when no camera snapshot/stream is available...

Reply:
> In Cameras: I hate the green image when no camera snapshot/stream is available...

That is on the list to improve in the 0.12.0 release, so I'd say leave it with the green screens for now as that is a separate PR which will address the green screen. Maybe having one message for cameras with errors and a different message for disabled cameras.

Reply:
I do not start camera processors and capture processes when a camera is disabled. Working. Will commit in the morning. Too tired now.

Naming looks strange:
start_camera_proces**sors** vs start_capture_proces**ses**
whereas inside camera_process and capture_process looks ok.

Reply:
> That is on the list to improve in the 0.12.0 release, so I'd say leave it with the green screens for now as that is a separate PR which will address the green screen. Maybe having one message for cameras with errors and a different message for disabled cameras.

I couldn't resist:
![Screenshot from 2022-10-27 10-39-01](https://user-images.githubusercontent.com/59219161/198235486-6fe839bf-f9c3-4331-81a0-79f00b45f57e.png)